streams/src/test/java/org/apache/kafka/streams/state/internals/StoreChangeLoggerTest.java

```diff
@@ -66,35 +68,41 @@ public String get(Integer key) {
         }
     };
 
+    @SuppressWarnings("unchecked")
     @Test
-    public void testAddRemove() {
+    public void testAddRemove() throws Exception {
+        Field dirtyField  = changeLogger.getClass().getDeclaredField("dirty");
+        Field removedField  = changeLogger.getClass().getDeclaredField("removed");
+        dirtyField.setAccessible(true);
+        removedField.setAccessible(true);
+
         context.setTime(1);
         written.put(0, "zero");
         changeLogger.add(0);
         written.put(1, "one");
         changeLogger.add(1);
         written.put(2, "two");
         changeLogger.add(2);
-        assertEquals(3, changeLogger.numDirty());
-        assertEquals(0, changeLogger.numRemoved());
+        assertEquals(3, ((Set<Integer>) dirtyField.get(changeLogger)).size());
```

Comment:
No strong opinion, but I'd probably just change the test to access the fields directly rather than through reflection (since they are visible from here).  Otherwise it's just weird.